### PR TITLE
Add ability to simulate offset wires

### PR DESCRIPTION
**Pull request by bonventre**
Uses new AlignedTrackerSim tables to add digi sim of offset wires.

Adds separate wire and straw position diagnostics to TrkStrawHitSeed/MC

Also changes LorentzAngle in Mu2eKinKal to be defined relative to drift direction from the POCA

### CODE REVIEW

File: TrackerMC/src/StrawDigisFromStrawGasSteps_module.cc [hunk 1]
```diff
@@ -1225,28 +1237,47 @@ namespace mu2e {
...
       XYZVectorF pdir = sdir.Cross(zdir); // radial direction
       if(pdir.Dot(smid) < 0.0)pdir *= -1.0; // sign radially outwards
       float dw = delta.Dot(sdir);
       XYZVectorF cperp = delta - dw*sdir; // just perp part
-      float phi = atan2(cperp.Dot(pdir),cperp.Dot(zdir));// angle around wire WRT Z axis in range -pi,pi
+      float phi = atan2(cperp.Dot(pdir),cperp.Dot(zdir)); // angle around wire WRT z axis in range -pi,pi
       float rho = min(sqrt(cperp.mag2()),(float)_rstraw); // truncate!
-      return StrawPosition(rho,dw,phi);
+      retval._strawPosition = StrawPosition(rho,dw,phi);
+      if (_simWireOffsets){
```
Comment (brownd1978) [inline TrackerMC/src/StrawDigisFromStrawGasSteps_module.cc:1253]: As remarked above, I would perefer we drive this behavior purely on database contents, ie always correct for wire offsets, but if the database purpose is 'perfect' those will be 0.

File: TrkDiag/src/StrawHitDiag_module.cc [hunk 2]
```diff
@@ -274,7 +274,7 @@ namespace mu2e
 
   void StrawHitDiag::fillStrawHitDiag(StrawElectronics const& strawele) {
     GeomHandle<DetectorSystem> det;
-    const Tracker& tracker = *GeomHandle<Tracker>();
+    const Tracker& tracker = *GeomHandle<Tracker>(); //FIXME switch to aligned
```
Comment (brownd1978) [inline TrkDiag/src/StrawHitDiag_module.cc:277]: There are a number of other downstream diagnostic modules where the digitization change needs to be propagated, ie ComboHitDiag and TrkAna.  Those changes are perhaps outside scope of this PR, but without them, how can you test and study the effects of this PR?

**Review by brownd1978 — COMMENTED** [review]
This is a really important PR adding to add the effect of wire offsets to the simulation.  The code looks good, but I would like to see some plots that verify that what it does corresponds to physical expectations.  It also seems that corresponding database content updates and updates to diagnostic modules are needed at the same time this merged.  Do you have a plan for how to coordinate those?

**Review by rlcee — CHANGES_REQUESTED** [review]
One small change requested, but important to set the example

File: TrkDiag/src/StrawHitDiag_module.cc (hunk 2 above)
Comment (bonventre) [reply TrkDiag/src/StrawHitDiag_module.cc:277]: mu2e/trkana#125 is ready for when this is merged, makeSD diag and combohitdiag both work. I'll be adding SHD asap

File: CosmicReco/src/CosmicTrackFit.cc [hunk 3]
```diff
@@ -168,13 +168,15 @@ namespace mu2e
         std::vector<StrawHitIndex> shitids;
         tseed._straw_chits.fillStrawHitIndices(ich, shitids);
 
+        /*
```
[the remark this answers is not in the document]
Comment (bonventre) [reply CosmicReco/src/CosmicTrackFit.cc:171]: TrkStrawHitSeeds were never used in CosmicTrackSeed, there is still cleanup to do in cosmicreco and it should eventually be removed

File: TrackerMC/src/MakeStrawGasSteps_module.cc [hunk 4]
```diff
@@ -441,7 +441,7 @@ namespace mu2e {
   }
 
   void MakeStrawGasSteps::fillStepDiag(Straw const& straw, StrawGasStep const& sgs, SPMCPV const& spmcptrs) {
-    _erad = sqrt((GenVector::Hep3Vec(sgs.endPosition())-straw.getMidPoint()).perpPart(straw.getDirection()).mag2());
+    _erad = sqrt((GenVector::Hep3Vec(sgs.endPosition())-straw.strawPosition()).perpPart(straw.strawDirection()).mag2());
```
[the remark this answers is not in the document]
Comment (bonventre) [reply TrackerMC/src/MakeStrawGasSteps_module.cc:444]: No it shouldn't change the output, this module still only uses the GeomHandle object

File: TrackerMC/src/MakeStrawGasSteps_module.cc [hunk 5]
```diff
@@ -456,7 +456,7 @@ namespace mu2e {
       _width = sgs.width();
       // compute DOCA to the wire
       TwoLinePCA poca(GenVector::Hep3Vec(sgs.startPosition()),GenVector::Hep3Vec(sgs.endPosition()-sgs.startPosition()),
-          straw.getMidPoint(),straw.getDirection());
+          straw.wirePosition(),straw.wireDirection());
```
[the remark this answers is not in the document]
Comment (bonventre) [reply TrackerMC/src/MakeStrawGasSteps_module.cc:459]: I'd prefer to remove the old interface from the straw object completely (it's currently marked as deprecated) so we don't confuse ourselves later where it does matter, here I'm just using the one that would make sense. Maybe that's more confusing though

File: TrackerConditions/src/AlignedTrackerMaker.cc [hunk 6]
```diff
@@ -50,8 +50,15 @@ namespace mu2e {
           std::array<xyzVec,2> wireends, strawends;
           for(int iend=0;iend < StrawEnd::nends; iend++){
             auto end = static_cast<StrawEnd::End>(iend);
-            auto wireend_UVW = ds_to_panel*straw.wireEnd(end) + straw_align.wireDeltaUVW(end);
-            auto strawend_UVW = ds_to_panel*straw.strawEnd(end) + straw_align.strawDeltaUVW(end);
+            auto wireend_UVW = ds_to_panel*straw.wireEnd(end);
```
[the remark this answers is not in the document]
Comment (bonventre) [reply TrackerConditions/src/AlignedTrackerMaker.cc:53]: I removed this and added a warning printout in case straw alignment is set here

**Comment by bonventre** [comment]
Put some slides showing example plots in docdb 48002. Since it only adds to the data products, TrkAna should work as is for now, PR is ready there as well for after this is merged

**Review by brownd1978 — APPROVED** [review]

**Comment by bonventre** [comment]
@rlcee Can you check that I've addressed your issue as well?

**Comment by kutschke** [comment]
@bonventre and @rlcee - I will watch this PR through the evening.  If you get the change made and Ray approves, i will rerun CI and merge tonight.

**Review by rlcee — APPROVED** [review]
thanks

**Comment by kutschke** [comment]
@FNALbuild run build test

**Pull request merged**